report/2.Analysis/EducationalMiniComputers.tex

```diff
@@ -0,0 +1,30 @@
+\section{Educational mini computers}
+
+Apart from the Arduino board, a lot of other mini computers have been created with educational purposes in mind.
+To figure out why these "mini computers" are so popular for technical learning, some of the most popular ones were analysed.
+This should help to figure out which functionality the HCL programming language should include in order to comply with the intend of the mini computers.
+
+\subsection{Raspberry Pi}
+The Raspberry Pi is one of the most widely known single board computers. 
+It is a complete arm-based computer, and may run anything arm based on it, including both linux and windows based arm operating systems. 
+This first of all makes it a common entry point into the GNU/Linux operating systsem stack, which is generally considered to be very educational in regards to how computers work.
+The default Raspberry Pi distribution ships with both scratch and python, allowing the user to quickly get started with some programming.\cite{RaspberryPi}
+
+On top of this, one of the major selling points to the Raspberry Pi is the GPIO\footnote{general purpose input output} pins. 
+These allow the user to attach all kind of devices to the board, this includes both sensors, lights, cameras and breadboards. 
+The Raspberry Pi allows the user to easily interact with the GPIO pins, creating a great device for both IoT, home automation and robotics.\cite{RaspberryPi}
+
+\subsection{Lego Mindstorm}
+Lego Mindstorm is a robot kit from lego. 
+The intend is to built a robot out of lego, with the belonging motors and sensors which can then be controlled by the "head". 
+The head is a small computer, which can control and monitor all the attached motors and sensors. 
+The computer can be programmed from either a tablet or a computer using a drag and drop programming environment similar to that of scratches (\ref{sec:scratch}). \cite{LegoMindstorms}
+
+\subsection{Takeaway}
+Apart from being cheap, the big motivation behind these mini computers seem to be the extensibility through attachments. 
+This allows the users to create something very concrete. 
+For instance, it may be more rewarding for the user to get an actual lamp to blink or a car to drive, rather than just outputting something on the computer screen. 
+The attachments will in the end allow the user to build something robotic, which is a common desire or goal for both new and experienced programmers.\cite{EducationalRobotics}
+
+% Uncertain wheter the below should be present here, or come later at some bigger "analysis conclusion"
```

Comment:
* I see no problem with having this conclusion here. Later it can easily be reapeated with much less exposition in the final analysis conclusion.